sdk/core/Azure.Core.TestFramework/src/RecordedTestSanitizer.cs

```diff
@@ -33,7 +37,24 @@ public virtual void SanitizeHeaders(IDictionary<string, string[]> headers)
 
         public virtual string SanitizeTextBody(string contentType, string body)
         {
-            return body;
+            if (JsonPathSanitizers.Count == 0)
+                return body;
+            try
+            {
+                var jsonO = JObject.Parse(body);
+                foreach (var jsonPath in JsonPathSanitizers)
+                {
+                    foreach (JToken token in jsonO.SelectTokens(jsonPath))
```

Comment:
You need to check if the contentType starts with "application/json" (not equals, since other information can follow).

Reply:
I think the current approach with fallback is fine, there are so many content type with JSON bodies that I would prefer not to try and parse them, especially if the cost of getting it wrong is leaking a secret.